lib/ansible/module_utils/ec2.py

```diff
@@ -592,7 +592,7 @@ def value_is_list(my_list):
 
         # Sort list. If it's a list of dictionaries, sort by tuple of key-value
         # pairs, since Python 3 doesn't allow comparisons such as `<` between dictionaries.
-        checked_list.sort(key=lambda x: sorted(x.items()) if isinstance(x, dict) else x)
+        checked_list.sort(key=lambda x: sorted(set(x.items())) if isinstance(x, dict) else x)
```

Comment:
Should this be 
``` python
checked_list.sort(key=lambda x: sorted(list(x.items())) if isinstance(x, dict) else x)
```

tracebacks from test failures in CI make me suspect the view object from set(x.items()) fails because set() wants hashable items and the view object from x.items() is not. Does wrapping it in a list() or tuple() work?
```
2017-06-26 16:43:33 The full traceback is:
2017-06-26 16:43:33 Traceback (most recent call last):
2017-06-26 16:43:33   File "/tmp/ansible_cLYNf7/ansible_module_ecs_ecr.py", line 304, in run
2017-06-26 16:43:33     policy = sort_json_policy_dict(policy)
2017-06-26 16:43:33   File "/tmp/ansible_cLYNf7/ansible_modlib.zip/ansible/module_utils/ec2.py", line 603, in sort_json_policy_dict
2017-06-26 16:43:33     ordered_policy_dict[key] = value_is_list(value)
2017-06-26 16:43:33   File "/tmp/ansible_cLYNf7/ansible_modlib.zip/ansible/module_utils/ec2.py", line 595, in value_is_list
2017-06-26 16:43:33     checked_list.sort(key=lambda x: sorted(set(x.items())) if isinstance(x, dict) else x)
2017-06-26 16:43:33   File "/tmp/ansible_cLYNf7/ansible_modlib.zip/ansible/module_utils/ec2.py", line 595, in <lambda>
2017-06-26 16:43:33     checked_list.sort(key=lambda x: sorted(set(x.items())) if isinstance(x, dict) else x)
2017-06-26 16:43:33 TypeError: unhashable type: 'list'
```